aibolit/patterns/non_final_class/non_final_class.py

```diff
@@ -11,9 +11,15 @@ def __init__(self):
         pass
 
     def value(self, filename: str) -> List[LineNumber]:
-        tree = build_ast(filename)
-        classes = tree.filter(ClassDeclaration)
-        return [
-            node.position.line for _, node in classes
-            if len([v for v in ['final', 'abstract'] if v in node.modifiers]) == 0
-        ]
+        tree = AST.build_from_javalang(build_ast(filename))
+        nodes = tree.nodes_by_type(ASTNodeType.CLASS_DECLARATION)
+
+        positions = []
+        for node in nodes:
+            line = tree.get_attr(node, 'source_code_line')
+            modifiers = tree.get_attr(node, 'modifiers')
+
+            if len([v for v in ['final', 'abstract'] if v in modifiers]) == 0:
```

Comment:
We should consider only main class, not inner classes. Is it right? In your example we count all inner classes